hotel_reservation/models/hotel_reservation.py

```diff
@@ -0,0 +1,1203 @@
+                and reserv_rec.partner_id.email
+                and reserv_rec.state == "confirm"
+            ):
+                template_id.send_mail(reserv_rec.id, force_send=True)
+        return True
+
+    def create_folio(self):
+        """
+        This method is for create new hotel folio.
+        -----------------------------------------
+        @param self: The object pointer
+        @return: new record set for hotel folio.
+        """
+        hotel_folio_obj = self.env["hotel.folio"]
+        for reservation in self:
+            folio_lines = []
+            checkin_date = reservation["checkin"]
+            checkout_date = reservation["checkout"]
+            if not self.checkin < self.checkout:
+                raise ValidationError(
+                    _(
+                        "Checkout date should be greater \
+                                         than the Check-in date."
+                    )
+                )
+            duration_vals = self.onchange_check_dates(
+                checkin_date=checkin_date,
+                checkout_date=checkout_date,
+                duration=False,
+            )
+            duration = duration_vals.get("duration") or 0.0
+            folio_vals = {
+                "date_order": reservation.date_order,
+                "warehouse_id": reservation.warehouse_id.id,
+                "partner_id": reservation.partner_id.id,
+                "pricelist_id": reservation.pricelist_id.id,
+                "partner_invoice_id": reservation.partner_invoice_id.id,
+                "partner_shipping_id": reservation.partner_shipping_id.id,
+                "checkin_date": reservation.checkin,
+                "checkout_date": reservation.checkout,
+                "duration": duration,
+                "reservation_id": reservation.id,
+            }
+            for line in reservation.reservation_line:
+                for r in line.reserve:
+                    folio_lines.append(
+                        (
+                            0,
+                            0,
+                            {
+                                "checkin_date": checkin_date,
+                                "checkout_date": checkout_date,
+                                "product_id": r.product_id and r.product_id.id,
+                                "name": reservation["reservation_no"],
+                                "price_unit": r.list_price,
+                                "product_uom_qty": duration,
+                                "is_reserved": True,
+                            },
+                        )
+                    )
+                    r.write({"status": "occupied", "isroom": False})
+            folio_vals.update({"room_lines": folio_lines})
+            folio = hotel_folio_obj.create(folio_vals)
+            if folio:
+                for rm_line in folio.room_lines:
+                    rm_line.product_id_change()
+            self._cr.execute(
```

Comment:
Can we exclude this extra sql if possible ?
I am sure many members won't agree for this